### PR TITLE
Enhance and clean up model documentation

### LINKED ISSUE #1620: Model documentation enhancement checklist
Following the merge of #1530, this meta-issue summarizes further enhancements that could, or should, be made to the model documentation: 

* [x] Avoid table scroll bars -> #1639
* [ ] Move ``Remarks:`` to description
* [ ] Remove remaining Doxygen syntax (e.g., ``@notes``)
* [X] ~Add additional column to parameter table (see #1605)~
  ➟ will be implemented by NESTML (see https://github.com/nest/nestml/issues/712)
* [ ] Update multiplication symbols (see #1605)
* [x] Rename ``spike_detector`` (depending on discussion outcome #1579)
* [x] Fix subscripts of ``ginzburg_neuron`` -> #1605
* [x] Remove devices from guides
* [x] Fix technical typos and spelling -> #1624

Ideally, this list will be split into a number of small PRs. 

@jougs Feel free to edit. @clinssen This might be interesting for you too.

### PR DESCRIPTION
**Pull request by jougs**
This PR addresses the remaining open action items and thereby fixes #1620.

### CODE REVIEW

**Comment by jougs** [comment]
@terhorstd, @jessica-mitchell: friendly ping!

File: models/amat2_psc_exp.h [hunk 1]
```diff
@@ -71,23 +71,17 @@ The general framework for the consistent formulation of systems with
...
-  - :math:`\tau_m != \tau_v`
-  This is required to avoid singularities in the numerics. This is a
-  problem of implementation only, not a principal problem of the model.
-- Expect unstable numerics if time constants that are required to be
-  different are very close.
-- :math:`\tau_m != \tau_{syn_{ex,in}}` is required by the current
-  implementation to avoid a degenerate case of the ODE describing the
-  model [1]_.  For very similar values, numerics will be unstable.
+The default parameter values for this model are different from the
+corresponding parameter values for mat2_psc_exp. If identical
```
Comment (jessica-mitchell) [inline models/amat2_psc_exp.h:75]:
```suggestion
corresponding parameter values for ``mat2_psc_exp``. If identical
```

File: models/cont_delay_synapse.h [hunk 2]
```diff
@@ -49,17 +49,15 @@ decomposed into an integer part (delay_) and a double (delay_offset_) so
...
 combined with off-grid spike times.
 
-Remarks:
-
 All delays set by the normal NEST Connect function will be rounded, even when
 using cont_delay_synapse. To set non-grid delays, you must either
 
-1) set the delay as synapse default, as in the example above
-2) set the delay for each synapse after the connections have been created,
+1. set the delay as model default using SetDefaults, which is very
```
Comment (jessica-mitchell) [inline models/cont_delay_synapse.h:55]:
```suggestion
1. set the delay as model default using ``SetDefaults``, which is very
```

File: models/iaf_psc_alpha_canon.h [hunk 3]
```diff
@@ -97,6 +97,18 @@ dynamics are integrated using exact integration between events [2]_.
    For implementation details see the
    `IAF_neurons_singularity <../model_details/IAF_neurons_singularity.ipynb>`_ notebook.
 
+This model transmits precise spike times to target nodes (on-grid spike
+time and offset). If this node is connected to a spike_recorder, the
```
Comment (jessica-mitchell) [inline models/iaf_psc_alpha_canon.h:101]:
```suggestion
time and offset). If this node is connected to a ``spike_recorder``, the
```

File: models/iaf_psc_alpha_canon.h [hunk 4]
```diff
@@ -97,6 +97,18 @@ dynamics are integrated using exact integration between events [2]_.
...
+property "precise_times" of the spike_recorder has to be set to true in
```
Comment (jessica-mitchell) [inline models/iaf_psc_alpha_canon.h:102]:
```suggestion
property "precise_times" of the ``spike_recorder`` has to be set to true in
```

File: models/iaf_psc_alpha_canon.h [hunk 5]
```diff
@@ -97,6 +97,18 @@ dynamics are integrated using exact integration between events [2]_.
...
+order to record the offsets in addition to the on-grid spike times.
+
+The iaf_psc_delta_ps neuron accepts connections transmitting
```
Comment (jessica-mitchell) [inline models/iaf_psc_alpha_canon.h:105]:
```suggestion
The ``iaf_psc_delta_ps`` neuron accepts connections transmitting
```

File: models/iaf_psc_alpha_canon.h [hunk 6]
```diff
@@ -97,6 +97,18 @@ dynamics are integrated using exact integration between events [2]_.
...
+CurrentEvents. These events transmit stepwise-constant currents which
```
Comment (jessica-mitchell) [inline models/iaf_psc_alpha_canon.h:106]:
```suggestion
``CurrentEvents``. These events transmit stepwise-constant currents which
```

File: models/iaf_psc_delta_ps.h [hunk 7]
```diff
@@ -92,6 +92,16 @@ of the nest simulation kernel because it is at the same time complex
 enough to exhibit non-trivial dynamics and simple enough compute
 relevant measures analytically.
 
+Please note that this node is capable of sending precise spike times
+to target nodes (on-grid spike time plus offset).
+
+The iaf_psc_delta_ps neuron accepts connections transmitting
```
Comment (jessica-mitchell) [inline models/iaf_psc_delta_ps.h:98]:
```suggestion
The ``af_psc_delta_ps`` neuron accepts connections transmitting
```

File: models/iaf_psc_delta_ps.h [hunk 8]
```diff
@@ -92,6 +92,16 @@ of the nest simulation kernel because it is at the same time complex
...
+CurrentEvents. These events transmit stepwise-constant currents which
```
Comment (jessica-mitchell) [inline models/iaf_psc_delta_ps.h:99]:
```suggestion
``CurrentEvents``. These events transmit stepwise-constant currents which
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
@jougs looks good, I just made some suggestions to mainly adding double back ticks in a few places

File: models/sinusoidal_poisson_generator.h [hunk 9]
```diff
@@ -54,24 +54,22 @@ The instantaneous rate of the process is given by
 .. math::
 
   f(t) = max(0, rate + amplitude \sin ( 2 \pi frequency t + phase
-     * \pi/180 )) >= 0
+     \cdot \pi/180 )) >= 0
```
[the remark this answers is not in the document]
Comment (jougs) [reply models/sinusoidal_poisson_generator.h:57]: I think the variable names should definitely be `mathrm`. Good catch. Let's see who gets to deal with the conflict resolution fallout ;-)

**Comment by jougs** [comment]
@jessica-mitchell: I have applied most of your suggestions and have two questions for the as-of-yet unapplied ones (mostly regarding `SetDefaults`):
* Instead of just making them typewriter by double-backticking them, what about marking the functions with `:py:func:`, so they are auto-linked to the PyNEST API?
* Is there a similar mechanism (existing/planned) for the models? The links into the model directory are also pretty much generic after all.

**Comment by jessica-mitchell** [comment]
>     * Instead of just making them typewriter by double-backticking them, what about marking the functions with `:py:func:`, so they are auto-linked to the PyNEST API?

Ah yes, that would make more sense, I just wasn't thinking about it 

>     * Is there a similar mechanism (existing/planned) for the models? The links into the model directory are also pretty much generic after all.

Yes there is a way to do this but not implemented in any way yet;i think this could be something we look at in a future PR though

File: models/stdp_dopamine_synapse.h [hunk 10]
```diff
@@ -101,18 +101,16 @@ Parameters
  Wmax      real    Maximal synaptic weight
 =========  ======= ======================================================
 
+The common properties can only be set by SetDefaults and apply to all
```
[the remark this answers is not in the document]
Comment (jougs) [reply models/stdp_dopamine_synapse.h]: Fixed in af4a0a15b.

File: models/stdp_synapse_facetshw_hom.h [hunk 11]
```diff
@@ -103,6 +110,8 @@ Parameters
                                     always reset) is allowed.
 ======================= =========== ===========================================
 
+Common properties can only be set on the synapse model using SetDefaults.
```
[the remark this answers is not in the document]
Comment (jougs) [reply models/stdp_synapse_facetshw_hom.h]: Fixed in af4a0a15b.

File: models/tsodyks_synapse_hom.h [hunk 12]
```diff
@@ -106,11 +106,11 @@ Parameters
...
 
-Remarks:
+.. note::
 
-The weight and the parameters U, tau_psc, tau_fac, and tau_rec are common to
-all synapses of the model and must be set using SetDefaults on the synapse
-model.
+   The weight and the parameters U, tau_psc, tau_fac, and tau_rec are
+   common to all synapses of the model and must be set using
+   SetDefaults on the synapse model.
```
[the remark this answers is not in the document]
Comment (jougs) [reply models/tsodyks_synapse_hom.h]: Fixed in af4a0a15b.

File: models/cont_delay_synapse.h (hunk 2 above)
Comment (jougs) [reply models/cont_delay_synapse.h]: Fixed in af4a0a15b.

**Pull request merged**